crates/system-env/src/helpers.rs

```diff
@@ -14,10 +14,7 @@ pub fn find_command_on_path<T: AsRef<OsStr>>(name: T) -> Option<PathBuf> {
     // Only extensions we care about
     let exts = vec![".exe", ".ps1", ".cmd", ".bat"];
     let name = name.as_ref();
-    let has_ext = name
-        .as_encoded_bytes()
```

Comment:
I originally went with this to avoid unnecessary string conversion.

Reply:
I think it's wrong in the case of UTF-16, where you're going through bytes instead of characters.

But also, the logic here is kinda iffy - on Windows, `PATHEXT` is appended to executable names. So `foo.bar` will look for `foo.bar.exe`, even though there is an extension there. Maybe we should just change it to `let exts = vec![".exe", ".ps1", ".cmd", ".bat", ""];` and skip the conditional?